sentry-android-core/src/main/java/io/sentry/android/core/AndroidSocketTagger.java

```diff
@@ -0,0 +1,27 @@
+package io.sentry.android.core;
+
+import android.net.TrafficStats;
+import io.sentry.ISocketTagger;
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+public final class AndroidSocketTagger implements ISocketTagger {
+
+  private static final AndroidSocketTagger instance = new AndroidSocketTagger();
+
+  private AndroidSocketTagger() {}
+
+  public static AndroidSocketTagger getInstance() {
+    return instance;
+  }
+
+  @Override
+  public void tagSockets() {
+    TrafficStats.setThreadStatsTag(1);
```

Comment:
Would it be better to have the current thread id as tag?
I see the value in not showing the violation, but if we want to tackle it, we can show the current thread id or a number hardly used by the user, to ensure we don't mess with any report of the app.
Also, we could use `TrafficStats.getAndSetThreadStatsTag()`, so that we can restore previous value, and if it's -1 clear it later. Not sure how much worth it is, as we are the only one using this thread, i think, but why not

Reply:
not sure if it'd change much because it's a threadlocal and we anyway use only our own single thread for our requests. But I changed it from 1 to a random number, I think it should be fine like that.